osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/deploymentspec/OsSvaServerCreateTask.java

```diff
@@ -131,53 +133,58 @@ public void executeTransaction(EntityManager em) throws Exception {
-        );
+                                           createdServer.getIngressInspectionMacAddr(),
+                                           createdServer.getIngressInspectionPortId(),
+                                           createdServer.getEgressInspectionMacAddr(),
+                                           createdServer.getEgressInspectionPortId());
+
         // Add new server ID to VM notification listener for this DS
 
         this.activeRunner.getOsDeploymentSpecNotificationRunner()
                 .addSVAIdToListener(this.dai.getDeploymentSpec().getId(), createdServer.getServerId());
 
-            if (vc.isControllerDefined()) {
-                try {
-                    DefaultNetworkPort ingressPort = new DefaultNetworkPort(createdServer.getIngressInspectionPortId(),
-                            createdServer.getIngressInspectionMacAddr());
-                    DefaultNetworkPort egressPort = new DefaultNetworkPort(createdServer.getEgressInspectionPortId(),
-                            createdServer.getEgressInspectionMacAddr());
-
-                    if (this.apiFactoryService.supportsPortGroup(this.dai.getVirtualSystem())) {
-                        String domainId = OpenstackUtil.extractDomainId(
-                                ds.getProjectId(),
-                                ds.getProjectName(),
-                                ds.getVirtualSystem().getVirtualizationConnector(),
-                                Arrays.asList(ingressPort));
-                        ingressPort.setParentId(domainId);
-                        egressPort.setParentId(domainId);
-                    }
-                    //Element object in DefaultInspectionport is not used at this point, hence null
-                    controller.registerInspectionPort(new DefaultInspectionPort(ingressPort, egressPort, null));
+        if (vc.isControllerDefined()) {
+            try {
+                DefaultNetworkPort ingressPort = new DefaultNetworkPort(createdServer.getIngressInspectionPortId(),
+                        createdServer.getIngressInspectionMacAddr());
+                DefaultNetworkPort egressPort = new DefaultNetworkPort(createdServer.getEgressInspectionPortId(),
+                        createdServer.getEgressInspectionMacAddr());
+
+                String redirectionTargetId = null;
+
+                if (this.apiFactoryService.supportsPortGroup(this.dai.getVirtualSystem())) {
+                    String domainId = OpenstackUtil.extractDomainId(ds.getProjectId(),
+                                                    ds.getProjectName(),
+                                                    ds.getVirtualSystem().getVirtualizationConnector(),
+                                                    Arrays.asList(ingressPort));
+
+                    ingressPort.setParentId(domainId);
+                    egressPort.setParentId(domainId);
+
+                    redirectionTargetId = ds.getPortGroupLabel();
```

Comment:
regardless of port group, this should be set to ds.getPortGroupLabel(). One of the reasons why term port group is confusing. It does not specific whether it is a group of ports of the workloads or the group of ports for inspection ports...